src/julia/tools.py

```diff
@@ -139,8 +140,11 @@ def julia_py_executable(executable=sys.executable):
     """
     Path to ``julia-py`` executable installed for this Python executable.
     """
-    stempath = os.path.join(os.path.dirname(executable), "julia-py")
-    candidates = {os.path.basename(p): p for p in glob.glob(stempath + "*")}
+    basedirs = {os.path.dirname(executable),
+                os.path.join(site.USER_BASE, "bin")}
```

Comment:
Why use set? I think we should use a list here in the order of increasing preference (so that the values of `candidates` are the one in the most preferred location).  Does `basedirs = [os.path.dirname(executable) + os.path.join(site.USER_BASE, "bin")] + site.PREFIXES` make sense?

Reply:
Actually, `os.path.join(site.USER_BASE, "bin")` may not work in Windows?  Is there a pre-defined path we can use?

Reply:
Thanks for the hint! I've changed the logics according to your remarks. On Windows there seem to be more difficulties anyway (starting with "libjulia" library extension), but this part looks like it should work if the rest does too.